### PR TITLE
feat: enable frontendless Function uploads

### PR DESCRIPTION
**Pull request by t-col**
- allows bundles to be uploaded without an intex.html UI entrypoint
- will still validate index.html UI entrypoints if they exist in a bundle
- will enforce validation of index.html if there are no Functions being uploaded
- brings contentful-app-scripts up to date with all available Function event types and templates

### CODE REVIEW

File: packages/contentful--create-contentful-app/src/includeFunction.ts [hunk 1]
```diff
@@ -12,10 +12,13 @@ const addBuildCommand = getAddBuildCommandFn({
 });
 
 const VALID_FUNCTION_TEMPLATES_DIRS = [
-  'external-references',
+  'appaction',
```
[the remark this answers is not in the document]
Comment (t-col) [reply packages/contentful--create-contentful-app/src/includeFunction.ts:15]: ah ok, thanks for the clarification!

[the remark this answers is not in the document]
Comment (t-col) [reply packages/contentful--create-contentful-app/src/includeFunction.ts]: https://github.com/contentful/create-contentful-app/pull/2223/commits/7d53c11ccd5e4581ab776a6f33e317a287f3fcb5

**Review by whitelisab — APPROVED** [review]
🚀

**Review by jjolton-contentful — APPROVED** [review]
❤️

**Pull request merged**